File: codewar/6-kyu/whichAreIn.js
// https://www.codewars.com/kata/550554fd08b86f84fe000a58/train/javascript


/**
 *
 * @param {Array} array1
 * @param {Array} array2
 */
function inArray(array1, array2) {
  let res = [];
  for (let i = 0; i < array2.length; i++) {
    for (let j = 0; j < array1.length; j++) {
      if (array2[i].includes(array1[j])) {
        res.push(array1[j]);
      }
    }
  }
  let removeDuplicate = [...new Set(res)];

  return !removeDuplicate.length ? [] : removeDuplicate.sort((a, b) => a > b);
}

let a1 = ["arp", "live", "strong"];
let a2 = ["lively", "alive", "harp", "sharp", "armstrong"];
console.log(inArray(a1, a2));

File: codewar/6-kyu/whichAreIn/WhichAreIn.go
package main

import (
	"fmt"
	"sort"
	"strings"
)

// https://www.codewars.com/kata/550554fd08b86f84fe000a58/train/go

// b/c codewar go version is old, just use `seen`
func contains(slice []string, val string) bool {
	for _, v := range slice {
		if v == val {
			return true
		}
	}
	return false
}

func InArray(array1 []string, array2 []string) []string {
	res := make([]string, 0)
	seen := make(map[string]bool)
	for _, word1 := range array1 {
		for _, word2 := range array2 {
			// if strings.Contains(word2, word1) && !slices.Contains(res, word1) {
			if strings.Contains(word2, word1) && !seen[word1] {
				seen[word1] = true
				res = append(res, word1)
			}
		}
	}
	sort.Strings(res)
	return res
}

func main() {
	a1 := []string{"arp", "live", "strong"}
	a2 := []string{"lively", "alive", "harp", "sharp", "armstrong"}
	fmt.Println(InArray(a1, a2))
}
